src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,469 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Animated, View, TouchableOpacity, TouchableHighlight} from 'react-native';
+import Interactable from 'react-native-interactable';
+import {BaseComponent, Constants, Colors, Typography} from '../../../src';
+
+
+const DEFAULT_HEIGHT = 72;
+const MIN_LEFT_MARGIN = 28;
+const DEFAULT_ICON_SIZE = 24;
+const ITEM_BG = Colors.blue30;
+const ITEM_PADDING = 12;
+const BLEED = 15;
+const ITEM_PROP_TYPES = {
+  width: PropTypes.number,
+  background: PropTypes.string,
+  text: PropTypes.string,
+  icon: PropTypes.number,
+  onPress: PropTypes.func,
+};
+
+/**
+ * @description: Interactable Drawer component
+ * @extendslink: 
+ */
+export default class Drawer extends BaseComponent {
+  static displayName = 'Drawer';
+
+  static propTypes = {
+    /**
+     * The drawer's height
+     */
+    height: PropTypes.number.isRequired,
+    /**
+     * The drawer's width
+     */
+    width: PropTypes.number,
+    /**
+     * The drawer top layer's damping
+     */
+    damping: PropTypes.number,
+    /**
+     * The drawer top layer's tention
+     */
+    tension: PropTypes.number,
+    /**
+     * The bottom layer's items to appear when opened from the right
+     */
+    rightItems: PropTypes.arrayOf(PropTypes.shape(ITEM_PROP_TYPES)),
+    /**
+     * The bottom layer's item to appear when opened from the left (a single item)
+     */
+    leftItem: PropTypes.shape(ITEM_PROP_TYPES),
+    /**
+     * The color for the text and icon tint of the items
+     */
+    itemsTintColor: PropTypes.string,
+    /**
+     * The items' icon size
+     */
+    itemsIconSize: PropTypes.number,
+    /**
+     * The items' text style
+     */
+    itemsTextStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+    /**
+     * Press handler (will also close the drawer)
+     */
+    onPress: PropTypes.func,
+  };
+
+  static defaultProps = {
+    height: DEFAULT_HEIGHT,
+    width: Constants.screenWidth,
+    damping: 1 - 0.6,
+    tension: 300,
+    itemsTintColor: Colors.white,
+    itemsIconSize: DEFAULT_ICON_SIZE,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.deltaX = new Animated.Value(0);
```

Comment:
if we repeat this code in `componentWillReceiveProps` I would have turned it into a separate method.

Reply:
Done.